app/utils/rule_based_recommender.py

```
@@ -40,7 +40,7 @@ def recommend(
         user: User,
         db_postgres: Session,
         db_mongo: Database,
-        top_k: int = 10,
+        top_k: int = None,  # None이면 전체 반환
```

Ruff (0.14.6)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)